src/components/Layout.tsx

```diff
@@ -70,11 +78,67 @@ interface NavBarProps {
   endpoint: GraphQLEndPoint;
 }
 export const NavBar: React.FC<NavBarProps> = ({ className, endpoint }) => {
-  const onSearch = (value: string) => {
+  const client = useApolloClient();
+  const onSearch = async (value: string) => {
+    value = value.trim();
     if (value.match(/^[0-9a-fA-F]{64}$/)) {
-      navigate(`/search/?${value}`);
+      try {
+        const data = await client.query<BlockByHashQuery>({
+          query: BlockByHashDocument,
+          variables: {hash: value},
+        });
```

Comment:
```suggestion
        const data = await client.query<BlockByHashQuery>({
          query: BlockByHashDocument,
          variables: {hash: value},
        }).catch(() => null);
```

Using `Promise.prototype.catch` would be more concise to read.

Reply:
Amended.